### ISSUE TITLE
new adaptor for `ona` with Oauth2

### ISSUE THREAD

**Issue by aleksa-krolls** [OpenFn/adaptors#580]
## Request
New adaptor to connect with `ona` API to: 
1. handle authentication using Oauth2 ([see docs](https://api.ona.io/static/docs/authentication.html#using-oauth2-with-the-ona-api))
2. generic `GET` helper so we can [get a list of forms](https://api.ona.io/static/docs/data.html#get-json-list-of-data-end-points), [paginate](https://api.ona.io/static/docs/data.html#paginate-data-of-a-specific-form), [get data for one submission](https://api.ona.io/static/docs/data.html#get-a-single-data-submission-for-a-given-form), and [query data points from 1 submission](https://api.ona.io/static/docs/data.html#query-submitted-data-of-a-specific-form)
3. generic `POST` to [add tags](https://api.ona.io/static/docs/data.html#tag-a-submission-data-point) to a submission

### Resources
- API docs: https://api.ona.io/static/docs/data.html
- Demo Login: we can access a demo proejct via Oauth using AK's gmail account - see LP `UNICEF Inform Demo`

## Customer considerations
No actual client work _yet_, but this would be really nice to have for an upcoming demo with UNICEF's `UNICARE` team on June 6th. If not possible, we'll make do without - but good to add to roadmap.

**Comment by aleksa-krolls** [comment]
@christad92 this is an adaptor that would be nice to have soon and that we have a demo credential for, but it requires Oauth. Let's discuss in our call tomorrow - not sure if all Oauth adaptors still need to go through product team, or if this is something Mtuchi/Hunter could deliver given the new generic Oauth implementation.

**Comment by christad92** [comment]
@aleksa-krolls from the call, I think Mtichi/Hunter can implement this regardless of the OAuth requirement. 

cc: @josephjclark

**Comment by aleksa-krolls** [comment]
@hunterachieng this is a new adaptor, so pls align with Joe on the approach before starting work next week :)

**Comment by hunterachieng** [comment]
@aleksa-krolls Do we have creds for Ona that I can use for authentication?

**Comment by josephjclark** [comment]
Hey, before starting this, do you think we could update the template https://github.com/OpenFn/adaptors/issues/583?

We can take the learnings from `satusehat`, implement them in the template, then run the template to do this adaptor. That should do 90% of the work "for free" and ensure good practice. We can then work out how to plug in oauth on top

If we can wait a day or two I may be able to update the template myself

**Comment by aleksa-krolls** [comment]
hey @josephjclark is that^ a question for me or @hunterachieng? Should @hunterachieng rather plug into the template #583 next?

**Comment by josephjclark** [comment]
Hey @aleksa-krolls this is done - ignore my comment please!

**Comment by aleksa-krolls** [comment]
@hunterachieng we're still blocked here, so we will wait until later in June to pick this back up. Keep you posted.

**Comment by martalovescoffee** [comment]
@aleksa-krolls are we still blocked?

**Comment by martalovescoffee** [comment]
Yes, still blocked until figure out how can we create an account.

**Comment by martalovescoffee** [comment]
@hunterachieng has contact.
